1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,48 @@
-function setAlarm() {}
+let flashParameter = 0;
+let flashIntervalBackground;
+
+function setAlarm() {
+  let timeOnScreen = document.getElementById("timeRemaining");
+  let inputTimeSeconds = document.getElementById("alarmSet");
+  let seconds = inputTimeSeconds.value;
+  inputTimeSeconds.value = "";
+  timeOnScreen.innerText = `Time Remaining: ${timeFormatter(seconds)}`;
+
+  let countDownInterval = setInterval(() => {
+    seconds--;
+    timeOnScreen.innerText = `Time Remaining: ${timeFormatter(seconds)}`;
+    if (timeOnScreen.innerText === "Time Remaining: 00:00") {
+      playAlarm();
+      clearInterval(countDownInterval);
+      flashInterval();
+    }
+  }, 1000);
+}
```

Comment:
This function works well, and the code is clear and concise . You could also consider making this into a function which is passed into setInterval() - think about the pros and cons of this.